src/lreq.h

```diff
@@ -27,6 +27,9 @@ typedef struct {
   void* data; /* extra data */
 } luv_req_t;
 
+// This is an arbitrary value that we can assume will never be returned by luaL_ref
+#define LUV_REQ_MULTIREF (-0x1234)
```

Comment:
Yeah, `LUA_NOREF` is `-2` and `LUA_REFNIL` is `-1` so I think this should be safe. Any real ref number will be positive (since the registry table is just a normal Lua array-like table).